### ISSUE TITLE
Pimp the tf.image to the tensorboard on validset in OnlineExport(callback)

### ISSUE THREAD

**Issue by tmquan** [ppwwyyxx/tensorpack#239]
Hi tensorpack, 

I would like to ask this issue:
I am using image2image.py and training with 2 datasets ds_train and ds_valid declared in 
```python
def get_config():
    logger.auto_set_dir()
    # dataset = get_data()
    ds_train, ds_valid = get_data()
    ds_train.reset_state()
    ds_valid.reset_state() 
    #print ds_train.size()
    return TrainConfig(
        dataflow=ds_train,
        callbacks=[
            PeriodicTrigger(ModelSaver(), every_k_epochs=100),
            ScheduledHyperParamSetter('learning_rate', [(200, 1e-4)]),
            PeriodicTrigger(InferenceRunner(ds_valid, [ScalarStats('L1_loss')]), every_k_epochs=1),
        ],
        model=Model(),
        steps_per_epoch=ds_train.size(),
        max_epoch=4000,
    )
```

The OnlineExport callback (http://tensorpack.readthedocs.io/en/latest/casestudies/colorize.html#callbacks) allows me to generate the input and predicted outputs as well as the related loss values but it doesn't show up similar tf.image in the tensorboard. 

Is there anyway to pimp the results of prediction on ds_valid to the tensorboard via this callback?

Best regards,

**Comment by ppwwyyxx** [comment]
There is a plan about it but not ongoing.
Basically the goal is to let the [monitor](https://github.com/ppwwyyxx/tensorpack/blob/master/tensorpack/callbacks/monitor.py) support data types other than scalar.

**Comment by Skylion007** [comment]
Also a request would be if you could allow the callback to say save images of the validation set in a folder. That would be extremely useful is one of the main issues I have with the current repository. Is there a way to hack inference runner to do this currently?

**Comment by ppwwyyxx** [comment]
If it is just a fixed validation set, why do you need to do it during training instead of before training?

And you can do that by writing an Inferencer. No need to hack anything.

**Comment by PatWie** [comment]
The new tensorboard has a slider below each image-summary. If `tf.summary.image` is called on the same input, one could inspect the result directly in the browser from different epochs.

**Comment by ppwwyyxx** [comment]
You can now use `self.trainer.monitors.put_image(name, array)` to put a numpy array to tensorboard.
